### ISSUE TITLE
Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED

### ISSUE THREAD

**Issue by QuaggWasTaken** [microsoft/vcpkg#6415]
I've been attempting to install CPPRestSDK by running .\vcpkg install cpprestsdk cpprestsdk:x64-windows, but I always get an error while building cpprestsdk:x86-windows. The error and logs are in the block below
My actual username in the log has been replaced with [My User] for privacy

```
 The following packages will be built and installed:

  * boost-algorithm[core]:x64-windows
  * boost-asio[core]:x64-windows
  * boost-atomic[core]:x64-windows
  * boost-bind[core]:x64-windows
  * boost-concept-check[core]:x64-windows
  * boost-container[core]:x64-windows
  * boost-context[core]:x64-windows
  * boost-conversion[core]:x64-windows
  * boost-coroutine[core]:x64-windows
  * boost-date-time[core]:x64-windows
  * boost-exception[core]:x64-windows
  * boost-function[core]:x64-windows
  * boost-function-types[core]:x64-windows
  * boost-functional[core]:x64-windows
  * boost-fusion[core]:x64-windows
  * boost-io[core]:x64-windows
  * boost-iterator[core]:x64-windows
  * boost-lambda[core]:x64-windows
  * boost-lexical-cast[core]:x64-windows
  * boost-math[core]:x64-windows
  * boost-numeric-conversion[core]:x64-windows
  * boost-optional[core]:x64-windows
  * boost-pool[core]:x64-windows
  * boost-range[core]:x64-windows
  * boost-regex[core]:x64-windows
  * boost-smart-ptr[core]:x64-windows
  * boost-thread[core]:x64-windows
  * boost-tokenizer[core]:x64-windows
  * boost-tuple[core]:x64-windows
  * boost-type-index[core]:x64-windows
  * boost-unordered[core]:x64-windows
  * brotli[core]:x64-windows
    cpprestsdk[brotli,core,default-features,websockets]:x86-windows
    cpprestsdk[brotli,core,default-features,websockets]:x64-windows
  * openssl[core]:x64-windows
  * openssl-windows[core]:x64-windows
  * websocketpp[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/38: cpprestsdk:x86-windows
Building package cpprestsdk[brotli,core,default-features,websockets]:x86-windows...
-- Using cached C:/Users/[My User]/CPPLibs/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
-- Using source at C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:202 (message):
    Command failed: C:/Program Files/CMake/bin/cmake.exe;--build;.;--config;Debug;--target;install;--;-v
    Working Directory: C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/x86-windows-dbg
    See logs for more information:
      C:\Users\[My User]\CPPLibs\vcpkg\buildtrees\cpprestsdk\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/cpprestsdk/portfile.cmake:43 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: cpprestsdk:x86-windows
  Vcpkg version: 2018.11.23-nohash
```

[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/3170389/install-x86-windows-dbg-out.log)
This is the log mentioned in the output that should contain extra information.

**Comment by Neumann-A** [comment]
`C:\Program Files (x86)\OpenSSL-Win32\lib\VC\libssl32MDd.lib C:\Program Files (x86)\OpenSSL-Win32\lib\VC\libcrypto32MDd.lib`

Linked against wrong openssl
probably related to #6394 will probably be solved by #6409 as soon has the regressions are fixed

strange that those lines from vcpkg.cmake seems not to work:
```
file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
set(CMAKE_SYSTEM_IGNORE_PATH
    "${_programfiles}/OpenSSL"
    "${_programfiles}/OpenSSL-Win32"
    "${_programfiles}/OpenSSL-Win64"
    "${_programfiles}/OpenSSL-Win32/lib/VC"
    "${_programfiles}/OpenSSL-Win64/lib/VC"
    "${_programfiles}/OpenSSL-Win32/lib/VC/static"
    "${_programfiles}/OpenSSL-Win64/lib/VC/static"
    "C:/OpenSSL/"
    "C:/OpenSSL-Win32/"
    "C:/OpenSSL-Win64/"
    "C:/OpenSSL-Win32/lib/VC"
    "C:/OpenSSL-Win64/lib/VC"
    "C:/OpenSSL-Win32/lib/VC/static"
    "C:/OpenSSL-Win64/lib/VC/static"
)

```

ah okay need also` PROGRAMFILES(x86)`

**Comment by Neumann-A** [comment]
@QuaggWasTaken: Please try out #6416 and #6409. Although #6416 will probably solve your problem it is likely that cpprestsdk is silently linking against the release openssl and not debug openssl. the linking issue is solved by #6409

**Comment by QuaggWasTaken** [comment]
I'm relatively new to this, so I'm not entirely sure of what all of that actually means. I'm very much a layman to vcpkg and cmake, so is there any good way to fix this, put in layman's terms? Sorry if this sounds needy or something, I just don't understand all of the terminology and stuff just yet.

**Comment by Neumann-A** [comment]
```
git checkout -B try_openssl_fix
git add remote Neumann-A  https://github.com/Neumann-A/vcpkg.git
git fetch Neumann-A 
git merge Neumann-A/patch-10
git merge Neumann-A/openssl_rename
vcpkg remove openssl-windows:x64-windows --recurse
```
retry to install cpprestsdk. Be aware that Neumann-A/openssl_rename still has regression so not everything might work.

**Comment by QuaggWasTaken** [comment]
The second command gave a fatal error of: fatal: pathspec 'remote' did not match any files

**Comment by Neumann-A** [comment]
ah okay should be 
`git remote add Neumann-A  https://github.com/Neumann-A/vcpkg.git`

**Comment by QuaggWasTaken** [comment]
Yeah, I googled the command and found that,  but it still didn't fix the install, still getting this

```
The following packages will be built and installed:
  * boost-asio[core]:x64-windows
  * brotli[core]:x64-windows
    cpprestsdk[brotli,core,default-features,websockets]:x86-windows
    cpprestsdk[brotli,core,default-features,websockets]:x64-windows
  * openssl[core]:x64-windows
  * openssl-windows[core]:x64-windows
  * websocketpp[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/7: cpprestsdk:x86-windows
Building package cpprestsdk[brotli,core,default-features,websockets]:x86-windows...
-- Using cached C:/Users/[My User]/CPPLibs/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
-- Using source at C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: ninja;-v
    Working Directory: C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\[My User]\CPPLibs\vcpkg\buildtrees\cpprestsdk\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:273 (vcpkg_execute_required_process)
  ports/cpprestsdk/portfile.cmake:29 (vcpkg_configure_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: cpprestsdk:x86-windows
  Vcpkg version: 2018.11.23-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by ras0219-msft** [comment]
This should hopefully be fixed by #6416 (thanks again @Neumann-A!).

@QuaggWasTaken Could you:
```
>vcpkg remove openssl --recurse
>vcpkg install cpprestsdk
```
and confirm that your issue is solved?

**Comment by QuaggWasTaken** [comment]
After using the 2 commands you provided, the install still errors out fatally. Here's the new error log:

```
The following packages will be built and installed:
    cpprestsdk[brotli,core,default-features,websockets]:x86-windows
Starting package 1/1: cpprestsdk:x86-windows
Building package cpprestsdk[brotli,core,default-features,websockets]:x86-windows...
-- Using cached C:/Users/[My User]/CPPLibs/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
-- Using source at C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: ninja;-v
    Working Directory: C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\[My User]\CPPLibs\vcpkg\buildtrees\cpprestsdk\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:273 (vcpkg_execute_required_process)
  ports/cpprestsdk/portfile.cmake:29 (vcpkg_configure_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: cpprestsdk:x86-windows
  Vcpkg version: 2018.11.23-nohash
```

**Comment by Neumann-A** [comment]
did you switch back to the master branch and pulled in the new version?

```
git checkout master
git pull
vcpkg remove openssl-windows --recurse
vcpkg install cpprestsdk
```

**Comment by JackBoosY** [comment]
@QuaggWasTaken, please delete folder _vcpkg/buildtrees/cpprestsdk_ and _vcpkg/buildtrees/openssl-windows_, use command `.\vcpkg.exe remove openssl-windows --recurse` and build cpprestsdk again.
Thanks.

**Comment by PhoebeHui** [comment]
@QuaggWasTaken, I tried to install cpprestsdk with latest vcpkg and vs2017 15.9.10, everything installed successfully. Please reopen it if it's still be an issue for you.

**Comment by zahera-fatima** [comment]
I need help please tell how to install rapidjson through vspkg.

I done this but it didn't work.
![image](https://user-images.githubusercontent.com/55752747/91251682-6468c880-e775-11ea-90eb-6a86c26b87ba.png)
